spec/services/invoices/preview_service_spec.rb

```diff
@@ -66,6 +66,39 @@
           end
         end
       end
+
+      context 'with applied coupons' do
+        let(:applied_coupon) do
+          build(
+            :applied_coupon,
+            customer: subscription.customer,
+            amount_cents: 2,
+            amount_currency: plan.amount_currency
+          )
+        end
+
+        it 'creates preview invoice for 2 days with applied coupons' do
+          travel_to(timestamp) do
+            result = described_class.new(customer:, subscription:, applied_coupons: [applied_coupon]).call
+
+            aggregate_failures do
```

Comment:
no need to aggregate failures. already aggregates by default.

Reply:
fixed on all places